### PR TITLE
Add custom character limit to pastes

### PR DESCRIPTION
**Pull request by csfore**
# Background

Recently I've been looking at alternatives to the service 0x0.st for pasting log files to. Logpaste seems like the perfect replacement, but it has limits on file size. Adding the ability for administrators to add custom file sizes would allow admins to make the file size the equivalent of 0x0's (512 MiB) or even larger. It would also allow them to set smaller limits as well if resources are limited or if the content being pasted doesn't warrant large sizes.

# Changes

- The default limit is now `2 * 1024 * 1024`, or 2 MiB instead of `2 * 1000 * 1000`, 2 MB
- Add cli flag `-maxsize int` to allow admins to set an arbitrary limit on paste file sizes if they would like to
  - The argument is passed in by # of MiB (so for a 512MiB limit, `-maxsize 512`, or for the default 2MiB, `-maxsize 2`)
- Add field `maxCharLimit` to the `defaultServer` struct for storing the given size
- Add fields to `handlers/paste_test.go` to use the new struct field in order for tests to pass
- Replace instances of `MaxPasteCharacters` with `s.maxCharLimit`

### CODE REVIEW

File: cmd/logpaste/main.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ func main() {
 		true, "whether to display usage information on homepage")
 	perMinuteLimit := flag.Int("perminutelimit",
 		0, "number of pastes to allow per IP per minute (set to 0 to disable rate limiting)")
+	maxCharLimit := flag.Int64("maxsize", 2, "max file size as MiB")
```
Comment (mtlynch) [inline cmd/logpaste/main.go:27]: Can we reword this to "max files size in MiB"?

And can we change the var name to `maxPasteMiB`? It's a little confusing to call it `maxCharLimit` at the point that it represents the MiB limit rather than the character limit.

The conversion from MiB to characters should be the same time we rename it like:

```golang
const charactersPerMiB = 1024 * 1024
maxCharLimit := maxPasteMiB * charactersPerMiB
```

**Review by mtlynch — CHANGES_REQUESTED** [review]
Cool, thanks for adding this! I just have a small note.

File: cmd/logpaste/main.go (hunk 1 above)
Comment (csfore) [reply cmd/logpaste/main.go:27]: Sounds good! For the other files, paste.go and server.go, is maxCharLimit fine in there or should I rename those too?

Comment (mtlynch) [reply cmd/logpaste/main.go:27]: Yeah, I think `main.go` is the only file that should deal in terms of max MiB. `main.go` should convert it to characters, and then the rest of the code deals in terms of max characters.

Comment (mtlynch) [reply cmd/logpaste/main.go:27]: Oh, can we also document the new flag in the README?

Comment (csfore) [reply cmd/logpaste/main.go:27]: Definitely, I'll be a little busy this weekend but hopefully I'll get those fixed by tomorrow evening!
